pkg/ruler/evaluator_jitter.go

```diff
@@ -2,21 +2,29 @@ package ruler
 
 import (
 	"context"
-	"math/rand"
+	"hash"
+	"math"
+	"sync"
 	"time"
 
 	"github.com/grafana/loki/pkg/logqlmodel"
 )
 
-// EvaluatorWithJitter wraps a given Evaluator. It applies a randomly-generated jitter (sleep) before each evaluation to
-// protect against thundering-herd scenarios where multiple rules are evaluated at the same time.
+// EvaluatorWithJitter wraps a given Evaluator. It applies a consistent jitter based on a rule's query string by hashing
+// the query string to produce a 32-bit unsigned integer. From this hash, we calculate a ratio between 0 and 1 and
+// multiply it by the configured max jitter. This ratio is used to delay evaluation by a consistent amount of random time.
+//
+// Consistent jitter is important because it allows rules to be evaluated on a regular, predictable cadence
+// while also ensuring that we spread evaluations across the configured jitter window to avoid resource contention scenarios.
 type EvaluatorWithJitter struct {
+	sync.Mutex
```

Comment:
nit: Not a big fan of this pattern of embedding mutex directly into struct particularly here. Since `EvaluatorWithJitter` is a public struct, you will have a public api `.Lock()` and `.Unlock()` methods on this struct, Meaning anyone can do locking and unlocking outside this package.

From the code below, I understand you want this lock accesible only inside private function `caculateJitter`.

Prefer to have something like `mu sync.Mutex` explicitly.